### ISSUE TITLE
Axes don't reach the end/is there a way to calibrate?

### ISSUE THREAD

**Issue by puyoxyz** [berarma/oversteer#68]
https://user-images.githubusercontent.com/15311104/123779604-a2250900-d8d2-11eb-8e26-507e245b9a54.mp4

Using a Logitech WingMan Formula Force GP, the axes don't reach the end. Because of this, I can't fully accelerate in ETS2 (I can only accelerate slowly)

On Windows I could fix this by calibrating the wheel in the Logitech software, but I couldn't find a calibration option anywhere in Oversteer

**Comment by berarma** [comment]
There's no graphical calibration tool for evdev devices as far as I know, and Overteer doesn't have that feature. Modern devices don't seem to need manual calibration.

You can try `evdev-joystick`, it's a console command that should be able to do calibration.

List your devices with: `evdev-joystick --l`

Request the current calibration settings with (change the path to the device according to the previous command): `evdev-joystick --s /dev/input/by-id/usb-Logitech_G29_Driving_Force_Racing_Wheel-event-joystick`

Running it without parameters will show some help to change the minimum and maximum values for any axis. You'll have to do some trial until you get the right values.

In case this works I could try to add this functionality to Oversteer but I don't know when. I advise you to put the commands in a script to calibrate it when you need to without having to find the right values again.

**Comment by berarma** [comment]
Did it work as I explained using `evdev-joystick`? I'd like to know since I may need to look at the way calibration is implemented there.

**Comment by isopix** [comment]
Or replace weared, cheap potentiometers by quality ones (or even
better by Hall Sensors. It would be then lifetime fix)

**Comment by berarma** [comment]
> Or replace weared, cheap potentiometers by quality ones (or even better by Hall Sensors. It would be then lifetime fix)
> […](#)
> On 6/29/21, Puyo ***@***.***> wrote: Closed #68. -- You are receiving this because you are subscribed to this thread. Reply to this email directly or view it on GitHub: [#68 (comment)](https://github.com/berarma/oversteer/issues/68#event-4953421013)

Please, keep it dead simple.

**Comment by parkerlreed** [comment]
This can be done with jstest-gtk in fewer steps.

Launch the program and select properties for the joystick. Calibrate > Start Calibration > Move all axis to the extreme and back to center > Done. I think that's remembered per session.

**Comment by puyoxyz** [comment]
> > This can be done with jstest-gtk in fewer steps.
> > Launch the program and select properties for the joystick. Calibrate > Start Calibration > Move all axis to the extreme and back to center > Done. I think that's remembered per session.
> 
> Jstest uses the joydev api so I don't think the calibration would take effect for any modern game.

Yep, I tried jstest-gtk before your comment and that didn't work (it worked in jstest-gtk, but not in games)



> Did it work as I explained using `evdev-joystick`? I'd like to know since I may need to look at the way calibration is implemented there.

Sorry, I got confused and thought I used evdev-joystick (I figured out how to calibrate before you replied) but I actually used [evdev-joystick-calibration](https://github.com/Virusmater/evdev-joystick-calibration) (the name is similar so when I saw your reply I thought it was the one I used)

**Comment by berarma** [comment]
> Sorry, I got confused and thought I used evdev-joystick (I figured out how to calibrate before you replied) but I actually used [evdev-joystick-calibration](https://github.com/Virusmater/evdev-joystick-calibration) (the name is similar so when I saw your reply I thought it was the one I used)

Thanks, I didn't know this tool. It seems to do the same but in a more user-friendly way. I'll take note.
